server/graphql/resolvers.ts

```diff
@@ -5,6 +5,12 @@ import { RequestTypeInterface } from '../models/requestTypeModel'
 import { ServerResponseInterface } from './serverResponse'
 import { Types } from 'mongoose'
 
+const softDeleteRequestTypeHelper = (id, dataSources): Promise<ServerResponseInterface> => {
+  const requestType = dataSources.requestTypes.getById(id)
+  requestType.requests.map(id => {dataSources.requests.softDelete(id)})
```

Comment:
We might need more error handling for softDelete
My recommendation: if `requests.softDelete(id)` doesn't succeed, then stop immediately and return `ServerResponseInterface` with error

Same for `RequestGroup` soft  delete. 

Thoughts?

Reply:
what does 'not succeed' mean? like if the id doesn't exist? or if the request is already deleted? or both

Reply:
if requests.softDelete(id) returns an error like
```
 {
          "success": false,
          "message": error._message,
          "id": id
        }
```

Reply:
I think a problem is that we shouldn't necessarily return a ServerResponseInterface from the datasource. Instead, simply return something similar to the Promise for the query for mutations. Then in the resolvers, create a corresponding ServerResponseInterface at the end of the resolver. In the body of the resolver, make calls to the datasource create/update/etc. as needed and keep track if any errors happened.

This way, the resolver can successfully manage things like: resolver needs to do (1) and (2), but if (1) fails then don't do (2), and if (2) fails,. then revert (1).

Reply:
^+1 for this suggestion
@kevinwang2040 what do u think?

Reply:
moving this convo onto slack...